### ISSUE TITLE
calling imitate multiple times

### ISSUE THREAD

**Issue by whitecolor** [staltz/xstream#138]
I wonder how `imitate` called multiple times should behave. Or it should be allowed to be called only once with an error thrown for example.
http://www.webpackbin.com/4kKOgnNxM

```js
import xs from 'xstream'

let interval1$ = xs.periodic(200).mapTo(1).debug('interval1$')

let interval2$ = xs.periodic(1000).mapTo(2)


let source$ = xs.create()

source$.imitate(interval1$)

setTimeout(() => {
  source$.imitate(interval2$)
}, 1000)

source$.take(10).addListener({
  next: (x) => console.log(x)
})
```

Though it could be interesting to allow multiple times imitate call with proper unsubscribing from previous source.

**Comment by staltz** [comment]
Imitate only exists to enable circular dependencies. It's unfortunately an imperative API, and I think it shouldn't be able to call it multiple times because then we're allowing dynamic circular dependencies, and this starts to get confusing very fast. We *could* make it through an error in that case, but it would only add more code to the bundle.

Let's keep in the list as an idea.

**Comment by whitecolor** [comment]
> but it would only add more code to the bundle.

Don't get me wrong, I know you care much about resulting bundle "size", but [this issue](https://github.com/cyclejs/cyclejs/issues/385): adds much more to current cycle bundles then potential couple lines of code 😉

**Comment by staltz** [comment]
I agree, and I'm still up for accepting a PR for that issue. The difference is that xstream is a much more foundational library than Cycle DOM. xstream is a dependency everywhere in Cycle.js, so its bundle size is more important.

**Comment by whitecolor** [comment]
> I agree, and I'm still up for accepting a PR for that issue. 

I think it (makeHTMLDriver) should be properly be extracted and documented maybe @TylorS could accomplish it. If no one will, I will someday 😄 

> xstream is a dependency everywhere in Cycle.js, so its bundle size is more important.

In **most real** cases it is supposed to be included only once per bundle, so it is not more important then any other component's size (esp dom driver for front-end web apps).

**Comment by staltz** [comment]
> In most real cases it is supposed to be included only once per bundle, so it is not more important then any other component's size (esp dom driver for front-end web apps). 

I didn't mean it would be included multiple times. I meant it will always be included, so it will determine the minimum size of any Cycle.js app. But anyway, we are not in disagreement.
